iohk/Appveyor.hs

```diff
@@ -0,0 +1,79 @@
+{-# OPTIONS_GHC -Weverything -Wno-unsafe -Wno-implicit-prelude -Wno-semigroup #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE FunctionalDependencies     #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE DataKinds                  #-}
+
+module Appveyor where
+
+import           Control.Lens.TH (makeFields)
+import           Data.Aeson      (FromJSON, parseJSON, withObject, (.:))
+import           Data.Monoid     ((<>))
+import           Data.String     (IsString)
+import qualified Data.Text       as T
+import           GHC.Generics    (Generic)
+import           Utils           (fetchJson)
+import           Types           (ApplicationVersion(ApplicationVersion), Arch(Win64))
+import           Data.Coerce                      (coerce)
+
+newtype JobId = JobId T.Text deriving (Show, Monoid)
+newtype BuildNumber = BuildNumber Integer deriving (Show)
+newtype Username = Username { usernameToText :: T.Text } deriving (Show, Monoid, IsString)
+newtype Project = Project { projectToText :: T.Text } deriving (Show, Monoid, IsString)
+
+data AppveyorBuild = AppveyorBuild {
```

Comment:
To make this a bit more informative, what about factoring out all those AppveyerBuildX types into:

```haskell
data AppveyorBuild = AppveyorBuild {
    _app_jobs :: [AppveyorJob]
 ,  _app_buildNumber :: BuildNumber
 , _app_version :: ApplicationVersion Win64
}

data AppveyorJob = AppveyorJob {
  _app_jobId :: JobId
}
```

(the use of the snake case for names is useful for `FromJSON`, as explained below).

Is there any subtle reason I'm missing? I think I can answer: we are forced to use whichever structure Appveyor is imposing on us, as we need to read the raw data back (using the `FromJSON` instances), but nevertheless I think we could benefit by the use of better names 😉